### PR TITLE
Re-structuring docs to have consistent sections

### PR DESCRIPTION
**Pull request by choldgraf**
This restructures our documentation to roughly follow the [divio documentation framework](https://documentation.divio.com/). It breaks down our docs into a few pieces:

- Tutorials (where `getting started` now lives)
- How To (now a top-level section)
- Reference
- About

The goal is to make the documentation easier / more consistent to navigate, and to make it more obvious how somebody could *extend* the documentation based on the kind of content they want to write.

I don't believe that it changes any of the URL structures, just the navigation in Sphinx (some pages are added / broken apart / etc however)

### CODE REVIEW

File: doc/index.md [hunk 1]
```diff
@@ -47,7 +52,15 @@ and often cover more conceptual topics.
 ```{toctree}
 :maxdepth: 2
 
-using/index
+howto/index
```
Comment (willingc) [inline doc/index.md:55]: I would change the title of this section to "How-to guides"

**Review by willingc — APPROVED** [review]
Looks great @choldgraf. I would make one title change and then ship it.

File: doc/using/using.rst [hunk 2]
```diff
@@ -12,48 +12,6 @@ own repository.
 For a more complete list of sample repositories for use with Binder, see the
 `Sample Binder Repositories <sample_repos.html>`_ page.
 
-Choose from many open source languages
```
Comment (betatim) [inline doc/using/using.rst:15]: Where did this information end up in the re-org?

Comment (choldgraf) [reply doc/using/using.rst:15]: Both of those topics are now covered in a how-to - good point that we don't want the top-level information to go away though. I've added a sentence linking to those guides in the landing page and on the "getting started" tutorial.

**Comment by choldgraf** [comment]
OK all - I think that I've addressed comments in the last couple of commits!

**Comment by choldgraf** [comment]
Hey all - if nobody objects, I'm going to resolve the merge conflict and then merge this in a couple of days

**Pull request merged**